pkg/docker/config/config_test.go

```diff
@@ -454,3 +459,56 @@ func TestGetAuthFailsOnBadInput(t *testing.T) {
 		t.Fatalf("expected JSON syntax error, not: %#+v", err)
 	}
 }
+
+func TestGetAllCredentials(t *testing.T) {
+	// Create a temporary authentication file.
+	tmpFile, err := ioutil.TempFile("", "auth.json.")
+	require.NoError(t, err)
+	_, err = tmpFile.Write([]byte{'{', '}'})
+	require.NoError(t, err)
+	err = tmpFile.Close()
+	require.NoError(t, err)
+	authFilePath := tmpFile.Name()
+	sys := types.SystemContext{AuthFilePath: authFilePath}
+
+	data := []struct {
+		server   string
+		username string
+		password string
+	}{
+		{
+			server:   "example.org",
+			username: "example-user",
+			password: "example-password",
+		},
+		{
+			server:   "quay.io",
+			username: "quay-user",
+			password: "quay-password",
+		},
+		{
+			server:   "localhost:5000",
+			username: "local-user",
+			password: "local-password",
+		},
+	}
+
+	// Write the credentials to the authfile.
+	for _, d := range data {
+		err := SetAuthentication(&sys, d.server, d.username, d.password)
+		require.NoError(t, err)
+	}
```

Comment:
(Absolutely non-blocking: There are already quite a few test files, including the legacy / non-normalized formats. Reading them would be more representative than creating a new config file using the latest `SetAuthentication` implementation.)